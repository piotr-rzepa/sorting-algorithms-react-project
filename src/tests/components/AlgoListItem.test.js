import * as React from 'react';
import { shallow } from 'enzyme';
import AlgorithmListItem from '../../components/AlgoListItem';
import { Link } from 'react-router-dom';
const data = [
	{
		id: 0,
		name: 'Bubble Sort',
		description:
			'Bubble sort, sometimes referred to as sinking sort, is a simple sorting algorithm',
		method: 'Exchanging',
		sortingTime: { best: 0, worst: 0, average: 10 }
	},
	{
		id: 1,
		name: 'Quick Sort',
		description: 'Quicksort is a divide-and-conquer algorithm.',
		method: 'Partitioning',
		sortingTime: { best: 0, worst: 0, average: 100 }
	}
];

describe('<AlgorithmListItem />', () => {
	const useContextMock = jest.spyOn(React, 'useContext');
	beforeEach(() => {
		useContextMock.mockClear();
	});

	it('renders correctly', () => {
		const wrapper = shallow(
			<AlgorithmListItem
				key={data[0].id}
				name={data[0].name}
				sortingTime={data[0].sortingTime}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('should choose correct theme', () => {
		useContextMock.mockReturnValue(true);
		const wrapper = shallow(
			<AlgorithmListItem
				key={data[0].id}
				name={data[0].name}
				sortingTime={data[0].sortingTime}
			/>
		);
		expect(wrapper.find(Link).get(0).props.style).toHaveProperty(
			'color',
			'white'
		);
	});
});
