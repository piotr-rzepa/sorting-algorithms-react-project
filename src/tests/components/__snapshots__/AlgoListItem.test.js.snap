// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlgorithmListItem /> renders correctly 1`] = `
<Fragment>
  <li>
    <Link
      style={
        Object {
          "color": "black",
        }
      }
      to="/algorithm/Bubble Sort"
    >
      Bubble Sort
    </Link>
    :
    10
  </li>
</Fragment>
`;
